sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -211,6 +211,15 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
       summary = getSummary(context))
   }
 
+  def invalidUTF8StringError(str: UTF8String): SparkIllegalArgumentException = {
+    new SparkIllegalArgumentException(
+      errorClass = "INVALID_UTF8_STRING",
+      messageParameters = Map(
+        "str" -> str.toString
```

Comment:
does `toString` always work for invalid utf8 string?

Reply:
Java's toString will always work for invalid utf8 string, but will sometimes use the Unicode replacement character � (U+FFFD) - I think this is fine for the purpose of diplaying the error message